pandas/core/strings/accessor.py

```diff
@@ -3036,13 +3038,31 @@ def _result_dtype(arr):
         return object
 
 
-def _get_single_group_name(rx):
-    try:
-        return list(rx.groupindex.keys()).pop()
-    except IndexError:
+def _get_single_group_name(regex: Pattern) -> Hashable:
+    if regex.groupindex:
+        return next(iter(regex.groupindex))
+    else:
         return None
 
 
+def _get_group_names(regex: Pattern) -> List[Hashable]:
```

Comment:
This code has been working for a while, but wondering if it'd be useful to add a test for this function, seems trivial to add.